src/bvh/util/bits.hpp

```diff
@@ -84,17 +84,13 @@ namespace bvh
 #error "unsupported compiler for find first set"
 #endif
   }
-  
-  inline std::uint32_t clz( std::uint32_t _val )
-  {
-    return _lzcnt_u32( _val );
-  }
-  
-  inline std::uint64_t clz( std::uint64_t _val )
+
+  template < typename T >
+  inline T clz( T _val )
   {
-    return _lzcnt_u64( _val );
+    return Kokkos::countl_zero( _val );
```

Comment:
For improved performance, we could use `Kokkos::Experimental::countl_zero_builtin` (it has fallback mechanism).

https://github.com/kokkos/kokkos/blob/8963927d056c1caf59931d972b5dcccace0191f8/core/src/Kokkos_BitManipulation.hpp#L419

Reply:
I think it's good for now how it is unless it turns out to be an issue